support-frontend/assets/components/priceCards/priceCardsContainer.tsx

```diff
@@ -33,43 +38,67 @@ const contributionTypeToPaymentInterval: Partial<
 };
 
 export function PriceCardsContainer({
-	frequency,
+	paymentFrequency,
 	renderPriceCards,
 }: PriceCardsContainerProps): JSX.Element {
 	const dispatch = useContributionsDispatch();
-	const currency = useContributionsSelector(
-		(state) => state.common.internationalisation.currencyId,
+	const { countryGroupId, currencyId } = useContributionsSelector(
+		(state) => state.common.internationalisation,
 	);
 	const { amounts } = useContributionsSelector((state) => state.common);
 	const { amountsCardData } = amounts;
 	const { selectedAmounts, otherAmounts } = useContributionsSelector(
 		(state) => state.page.checkoutForm.product,
 	);
 	const minAmount = useContributionsSelector(getMinimumContributionAmount());
+
+	const { abParticipations } = useContributionsSelector(
+		(state) => state.common,
+	);
+	const inThreeTierVariantB = abParticipations.threeTierCheckout === 'variantB';
+
+	const getTierLowPiceCardAmounts = (
+		contributionType: ContributionType,
+		countryGroup: CountryGroupId,
+		intThreeTierVariant: boolean,
+	): AmountValuesObject | undefined => {
+		const tierBillingPeriod =
+			contributionType === 'ANNUAL' ? 'annual' : 'monthly';
+		const priceCards = tierCards[`tier1`].plans[tierBillingPeriod].priceCards;
+		if (priceCards && intThreeTierVariant) {
+			return priceCards[countryGroup];
+		}
+	};
+
 	const {
 		amounts: frequencyAmounts,
 		defaultAmount,
 		hideChooseYourAmount,
-	} = amountsCardData[frequency];
+	} = getTierLowPiceCardAmounts(
+		paymentFrequency,
+		countryGroupId,
+		inThreeTierVariantB,
+	) ?? amountsCardData[paymentFrequency];
```

Comment:
To keep the three tier logic out of the method and leave it to the implementing component, you could do something like:

```suggestion
	} = inThreeTierVariantB ? getTierLowPiceCardAmounts(
		paymentFrequency,
		countryGroupId,
		) : amountsCardData[paymentFrequency];
```

That way `getTierLowPiceCardAmounts` would always return something.

Reply:
Am not sure this works because the call `getTierLowPriceCardsAmounts` can also legitimately  return `undefined` and you would get an type error in this scenario with the solution above.

Reply:
Using Georges suggestion below,  I have simplified and removed `getTierLowPriceCardAmounts` as requested with this rewrite:
```
	const inThreeTierVariantB = inThreeTierV2VariantB(
		useContributionsSelector((state) => state.common.abParticipations),
	);
	const tierBillingPeriod =
		paymentFrequency === 'ANNUAL' ? 'annual' : 'monthly';
	const tierCardData =
		tierCardsVariantB.tier1.plans[tierBillingPeriod].priceCards;
	const {
		amounts: frequencyAmounts,
		defaultAmount,
		hideChooseYourAmount,
	} = inThreeTierVariantB && tierCardData
		? tierCardData[countryGroupId]
		: amountsCardData[paymentFrequency];
```